students/hyska_monika/lesson_06_dicts_tuples_sets_args_kwargs/GuessTheNumber.py

```diff
@@ -0,0 +1,29 @@
+# Program return numbers that can are a secret
+# (after receive yes or no answer for lists)
+from functions_lesson06 import create_dict_to_n, find_max
+
+
+def number_check(asked_numbers):
+    while True:
+        numbers = (input("put numbers with space: "))
+        if numbers == "help":
+            asked_numbers = {key: val for key, val in asked_numbers.items()
+                             if val != 0}
+            break
+        else:
+            numbers = numbers.split(' ')
+            numbers = map(int, numbers)
+            answer = (input("yes/no: "))
+            if answer == "yes":
+                for elem in numbers:
```

Comment:
Now this is a place, where `set()` would be great. Something like this (I'm reading the HELP command in the wrong place, but you get the idea):
```
valid = set(range(1,n))
row = set(map(int, input().split(' '))
command = input()
if command == 'YES':
  valid &= row
elif command == 'NO':
  valid -= row
elif command == 'HELP':
   print(valid)
else:
  raise ValueError('Bad command')
```
The clue that you are using the wrong tool is that you have to do many `for` loops.